src/erlcloud_sns.erl

```diff
@@ -580,12 +580,15 @@ publish(Type, RecipientArn, Message, Subject, Attributes, Config) ->
             Subject -> [{"Subject", Subject}]
         end,
     AttributesParam = message_attributes(Attributes),
-    Doc =
-        sns_xml_request(
+    try sns_xml_request(
             Config, "Publish",
-            RecipientParam ++ MessageParams ++ SubjectParam ++ AttributesParam),
-    erlcloud_xml:get_text(
-        "PublishResult/MessageId", Doc).
+            RecipientParam ++ MessageParams ++ SubjectParam ++ AttributesParam) of
+        Doc ->
+            erlcloud_xml:get_text("PublishResult/MessageId", Doc)
+    catch
+        exit:{fatal,{expected_element_start_tag, _, _, _}}:_Stack ->
+            erlang:error({sns_error, invalid_xml_response_document})
```

Comment:
I would really like this to return the raw response, like `{aws_error, {invalid_xml_response_document, Response}}`. Is it possible to catch the above error farther up where we can collect this response?

I don't think anyone at the center of all this will care about us handling this exception and returning something generic that isn't garbage and clearly points to the real problem.

Reply:
Updated to handle invalid xml in `erlcloud_aws` enabling a generic error return containing the invalid body